### ISSUE TITLE
observeGroupsAndScenes hangs after gateway update

### ISSUE THREAD

**Issue by lpgera** [AlCalzone/node-tradfri-client#437]
Whenever I call the `observeGroupsAndScenes()` method, it just hangs and never returns.

I can discover my gateway, connect to it, observe devices and interact with them without any issues, so the issue is either around groups or scenes.

I guess this has something to do with the new app and Gateway update, which advertises that it replaces the old Moods with the new Scenes. My Gateway firmware version is 1.12.31.

<details>
  <summary>Attaching logs</summary>
  
```
[debug] Attempting connection. Identity = [redacted], psk = [redacted]
[debug] Connection successful
[info] got all groups: [131079,131078,131080,131075,131077,131076,131074]
[debug] adding groups with keys [131074,131075,131076,131077,131078,131079,131080]
[debug] removing groups with keys []
[silly] Group: checking if name is required...
[silly] Group: checking if createdAt is required...
[silly] Group: checking if instanceId is required...
[silly] Group: checking if onOff is required...
[silly] Group: checking if dimmer is required...
[silly] Group: checking if deviceIDs is required...
[silly] Group: checking if sceneId is required...
[silly] Group: checking if groupType is required...
[silly] Group: checking if name is required...
[silly] Group: checking if createdAt is required...
[silly] Group: checking if instanceId is required...
[silly] Group: checking if onOff is required...
[silly] Group: checking if dimmer is required...
[silly] Group: checking if deviceIDs is required...
[silly] Group: checking if sceneId is required...
[silly] Group: checking if groupType is required...
[silly] Group: checking if name is required...
[silly] Group: checking if createdAt is required...
[silly] Group: checking if instanceId is required...
[silly] Group: checking if onOff is required...
[silly] Group: checking if dimmer is required...
[silly] Group: checking if deviceIDs is required...
[silly] Group: checking if sceneId is required...
[silly] Group: checking if groupType is required...
[silly] Group: checking if name is required...
[silly] Group: checking if createdAt is required...
[silly] Group: checking if instanceId is required...
[silly] Group: checking if onOff is required...
[silly] Group: checking if dimmer is required...
[silly] Group: checking if deviceIDs is required...
[silly] Group: checking if sceneId is required...
[silly] Group: checking if groupType is required...
[silly] Group: checking if name is required...
[silly] Group: checking if createdAt is required...
[silly] Group: checking if instanceId is required...
[silly] Group: checking if onOff is required...
[silly] Group: checking if dimmer is required...
[silly] Group: checking if deviceIDs is required...
[silly] Group: checking if sceneId is required...
[silly] Group: checking if groupType is required...
[silly] Group: checking if name is required...
[silly] Group: checking if createdAt is required...
[silly] Group: checking if instanceId is required...
[silly] Group: checking if onOff is required...
[silly] Group: checking if dimmer is required...
[silly] Group: checking if deviceIDs is required...
[silly] Group: checking if sceneId is required...
[silly] Group: checking if groupType is required...
[silly] Group: checking if name is required...
[silly] Group: checking if createdAt is required...
[silly] Group: checking if instanceId is required...
[silly] Group: checking if onOff is required...
[silly] Group: checking if dimmer is required...
[silly] Group: checking if deviceIDs is required...
[silly] Group: checking if sceneId is required...
[silly] Group: checking if groupType is required...
[info] got all scenes in group 131074: [196631,196632]
[debug] adding scenes with keys [196631,196632] to group 131074
[debug] removing scenes with keys [] from group 131074
[info] got all scenes in group 131075: []
[debug] adding scenes with keys [] to group 131075
[debug] removing scenes with keys [] from group 131075
[info] got all scenes in group 131076: []
[debug] adding scenes with keys [] to group 131076
[debug] removing scenes with keys [] from group 131076
[info] got all scenes in group 131077: []
[debug] adding scenes with keys [] to group 131077
[debug] removing scenes with keys [] from group 131077
[info] got all scenes in group 131078: []
[debug] adding scenes with keys [] to group 131078
[debug] removing scenes with keys [] from group 131078
[info] got all scenes in group 131079: [196633,196634]
[debug] adding scenes with keys [196633,196634] to group 131079
[debug] removing scenes with keys [] from group 131079
[info] got all scenes in group 131080: [196635,196636,196637]
[debug] adding scenes with keys [196635,196636,196637] to group 131080
[debug] removing scenes with keys [] from group 131080
[warn] found unknown property with key 9109
[warn] object was: {"9001":"Night","9002":1603310921,"9003":196631,"9057":3,"9068":4,"9109":0,"15013":[{"5706":"0","5707":5441,"5708":42889,"5709":30087,"5710":26909,"5711":0,"5850":0,"5851":1,"9003":65539},{"5706":"f1e0b5","5707":5427,"5708":42596,"5709":30015,"5710":26870,"5711":0,"5850":0,"5851":1,"9003":65538},{"5706":"0","5707":5441,"5708":42889,"5709":30087,"5710":26909,"5711":0,"5850":0,"5851":2,"9003":65540}]}
[silly] Scene: checking if name is required...
[silly] Scene: checking if createdAt is required...
[silly] Scene: checking if instanceId is required...
[silly] Scene: checking if isActive is required...
[silly] Scene: checking if isPredefined is required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if sceneIndex is required...
[silly] Scene: checking if useCurrentLightSettings is required...
[warn] found unknown property with key 9109
[warn] object was: {"9001":"Day","9002":1603310977,"9003":196632,"9057":4,"9068":4,"9109":0,"15013":[{"5706":"0","5707":5441,"5708":42889,"5709":30087,"5710":26909,"5711":0,"5850":0,"5851":254,"9003":65539},{"5706":"f1e0b5","5707":5427,"5708":42596,"5709":30015,"5710":26870,"5711":0,"5850":0,"5851":254,"9003":65538},{"5706":"0","5707":5441,"5708":42889,"5709":30087,"5710":26909,"5711":0,"5850":0,"5851":254,"9003":65540}]}
[silly] Scene: checking if name is required...
[silly] Scene: checking if createdAt is required...
[silly] Scene: checking if instanceId is required...
[silly] Scene: checking if isActive is required...
[silly] Scene: checking if isPredefined is required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if sceneIndex is required...
[silly] Scene: checking if useCurrentLightSettings is required...
[warn] found unknown property with key 9109
[warn] object was: {"9001":"Night","9002":1603311061,"9003":196633,"9057":3,"9068":4,"9109":0,"15013":[{"5706":"f1e0b5","5707":5429,"5708":43032,"5709":30138,"5710":26909,"5711":0,"5850":0,"5851":2,"9003":65551},{"5706":"0","5707":5441,"5708":42889,"5709":30087,"5710":26909,"5711":0,"5850":0,"5851":2,"9003":65553},{"5706":"0","5707":5441,"5708":42889,"5709":30087,"5710":26909,"5711":0,"5850":0,"5851":2,"9003":65552}]}
[silly] Scene: checking if name is required...
[silly] Scene: checking if createdAt is required...
[silly] Scene: checking if instanceId is required...
[silly] Scene: checking if isActive is required...
[silly] Scene: checking if isPredefined is required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if sceneIndex is required...
[silly] Scene: checking if useCurrentLightSettings is required...
[warn] found unknown property with key 9109
[warn] object was: {"9001":"Day","9002":1603311078,"9003":196634,"9057":4,"9068":4,"9109":0,"15013":[{"5706":"f1e0b5","5707":5429,"5708":43032,"5709":30138,"5710":26909,"5711":0,"5850":0,"5851":254,"9003":65551},{"5706":"0","5707":5441,"5708":42889,"5709":30087,"5710":26909,"5711":0,"5850":0,"5851":254,"9003":65553},{"5706":"0","5707":5441,"5708":42889,"5709":30087,"5710":26909,"5711":0,"5850":0,"5851":254,"9003":65552}]}
[silly] Scene: checking if name is required...
[silly] Scene: checking if createdAt is required...
[silly] Scene: checking if instanceId is required...
[silly] Scene: checking if isActive is required...
[silly] Scene: checking if isPredefined is required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if sceneIndex is required...
[silly] Scene: checking if useCurrentLightSettings is required...
[warn] found unknown property with key 9109
[warn] object was: {"9001":"ALLOFF","9002":44,"9003":196635,"9057":0,"9068":3,"9109":0,"15013":[{"5850":0,"9003":65539},{"5850":0,"9003":65548},{"5850":0,"9003":65538},{"5850":0,"9003":65544},{"5850":0,"9003":65542},{"5850":0,"9003":65553},{"5850":0,"9003":65543},{"5850":0,"9003":65551},{"5850":0,"9003":65540},{"5850":0,"9003":65552}]}
[silly] Scene: checking if name is required...
[silly] Scene: checking if createdAt is required...
[silly] Scene: checking if instanceId is required...
[silly] Scene: checking if isActive is required...
[silly] Scene: checking if isPredefined is required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if sceneIndex is required...
[silly] Scene: checking if useCurrentLightSettings is required...
[warn] found unknown property with key 9109
[warn] object was: {"9001":"L Night","9002":1606341230,"9003":196636,"9057":1,"9068":2,"9109":0,"15013":[{"5850":0,"9003":65542},{"5850":0,"9003":65543},{"5850":0,"9003":65544}]}
[silly] Scene: checking if name is required...
[silly] Scene: checking if createdAt is required...
[silly] Scene: checking if instanceId is required...
[silly] Scene: checking if isActive is required...
[silly] Scene: checking if isPredefined is required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if sceneIndex is required...
[silly] Scene: checking if useCurrentLightSettings is required...
[warn] found unknown property with key 9109
[warn] object was: {"9001":"Day","9002":1606342100,"9003":196637,"9057":2,"9068":2,"9109":0,"15013":[{"5850":0,"9003":65542},{"5850":0,"9003":65543},{"5850":0,"9003":65544}]}
[silly] Scene: checking if name is required...
[silly] Scene: checking if createdAt is required...
[silly] Scene: checking if instanceId is required...
[silly] Scene: checking if isActive is required...
[silly] Scene: checking if isPredefined is required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if lightSettings is required...
[silly] LightSetting: checking if name is required...
[silly] LightSetting: checking if createdAt is required...
[silly] LightSetting: checking if instanceId is required...
[silly] LightSetting: checking if hue is required...
[silly] LightSetting: checking if saturation is required...
[silly] LightSetting: checking if colorTemperature is required...
[silly] LightSetting: checking if dimmer is required...
[silly] LightSetting: checking if onOff is required...
[silly] LightSetting: checking if hue is always required...
[silly] Scene: checking if sceneIndex is required...
[silly] Scene: checking if useCurrentLightSettings is required...
[debug] ping succeeded
[debug] setting next timeout in 10000
[debug] ping succeeded
[debug] setting next timeout in 10000
[debug] ping succeeded
[debug] setting next timeout in 10000
[debug] ping succeeded
[debug] setting next timeout in 10000
```
  
</details>

Let me know if there's more information needed!

**Comment by AlCalzone** [comment]
> I guess this has something to do with the new app and Gateway update, which advertises that it replaces the old Moods with the new Scenes

That could be it. Could you create me a tcpdump of the communication? (I'll need your security code and/or credentials to decrypt it though)
https://github.com/AlCalzone/node-tradfri-client/issues/351#issuecomment-588197885

**Comment by lpgera** [comment]
> I'll need your security code and/or credentials to decrypt it though

I'd be more comfortable with the data being decrypted on my end. I've never done this with DTLS though, can it be done in Wireshark, or do I need some kind of other tool for it?

**Comment by AlCalzone** [comment]
Yep, you can decrypt the data using Wireshark. You need to enter the security key (converted from ascii to hex) as the PSK in the corresponding dialog.

**Comment by lpgera** [comment]
Thanks! Here's the decrypted capture file:

[decrypted.pcapng.zip](https://github.com/AlCalzone/node-tradfri-client/files/5603343/decrypted.pcapng.zip)

**Comment by lpgera** [comment]
I still couldn't get to the bottom of this, but I think the issue is caused by [this](https://github.com/AlCalzone/node-tradfri-client/blob/master/src/tradfri-client.ts#L597) `Promise.all` never resolving. Looking into it further I discovered that those `observeScenesPromises` never resolve which belong to groups without any scenes. I believe this never happened before, but with the new IKEA update the predefined scenes are now gone.

**Comment by AlCalzone** [comment]
FYI, I hope I'll have some time tomorrow to look at this in depth.

**Comment by jenskastensson** [comment]
@AlCalzone as an alternative to the sponsorship, is it possible to make single donations?

**Comment by AlCalzone** [comment]
@jenskastensson I've added PayPal to the sponsor links

**Comment by AlCalzone** [comment]
@lpgera it doesn't seem to have to do with the update or the gateway version.
I've just tested on mine which also has the version `1.12.31`:
```
	// connection and stuff
	await tradfri.observeDevices();
	await tradfri.observeGroupsAndScenes();
	console.log("done awaiting groups and scenes")
```

console output:
```
device with ID 65536 updated...
device with ID 65537 updated...
device with ID 65538 updated...
device with ID 65539 updated...
device with ID 65540 updated...
device with ID 65541 updated...
done awaiting groups and scenes
```

Your dump also seems to be fine:
1. It enumerates the groups (7 total)
2. It requests info about each group (7 calls total, all resolve)
3. It requests the scenes of each group (7 calls total, all resolve, only 3 have scenes (2+2+3))
4. It requests each of the scenes (7 calls total, all resolve)
5. silence...

My guess is that this behavior might be related to some groups having no scenes. However I can't reproduce it and thus cannot debug.
Would it be possible for you to set a breakpoint in `tradfri-client.ts`, line 723 (inside observeScenePromises > handleResponse) and check if you find something off?

**Comment by sergeykhomyuk** [comment]
Hello, 

I'm getting the same issue: promise returned by `TradfriClient.observeGroupsAndScenes()` is never resolved.
I debugged it a bit and could confirm that issue is caused by the fact that some groups don't have scenes.
To be precise: only "SuperGroup" (`groupType: 2`) has scenes, all other manually created groups (`groupType: 0`) has no scenes at all. 

```
got all scenes in group 131073: [196608,196609]
got all scenes in group 131074: []
got all scenes in group 131075: []
got all scenes in group 131076: []
```

I have v1.12.31 on Tradfri as well. You probably need to reboot your hub to reproduce this issue. 
I got this issue after I rebooted my hub yesterday.

**Comment by lpgera** [comment]
It looks like `handleResponse` in `observeScenePromises` gets called for every scene, thus it does not get called for those groups which have no scenes. And each group's promise can get resolved only from inside this function, so those groups which have no scenes always get stuck in this pending promise state.

If I add this block above the `const observeScenePromises = ...` line, it seems to do the trick:

```
if (newKeys.length === 0) {
	this.observeScenesPromises?.get(groupId)?.resolve();
}
```

But I'm not sure whether it breaks such things as listening to scene changes in these groups.

**Comment by AlCalzone** [comment]
@lpgera That does the trick. I could reproduce with a test case and with a (very similar) fix it continued without stalling.

**Comment by AlCalzone** [comment]
fixed in v2.1.8 (available in a couple of minutes)

**Comment by lpgera** [comment]
Thank you @AlCalzone, I really appreciate your work on this library!
